great_expectations/rule_based_profiler/config/base.py

```diff
@@ -51,21 +51,36 @@ def make_config(self, data: dict, **kwargs) -> Type[DictDot]:
 
         return self.__config_class__(**data)
 
-    @post_dump
-    def remove_nulls(self, data: dict, **kwargs) -> dict:
+    @post_dump(pass_original=True)
+    def remove_nulls_and_keep_unknowns(
```

Comment:
This is a very much needed piece of Marshmallow magic.

Since we use `setattr` to dynamically set values, the schema isn't aware of all possible types the object can have. When writing to a dictionary using `dump`, the config object validates against the schema and then serializes, implicitly removing unknown values as part of the process.

We want to persist these values - as such, we use the `pass_original=True` argument in the decorator to gain access to the original config object. From here, we can ensure that we're persisting this unknown values in the resulting dictionary object.

One caveat is that we need to manually filter our private attributes (like `_commented_map`) to get this working. Perhaps this is a little hacky but it follows Marshmallow conventions and is the cleanest way to enable proper serialization when utilizing our `setattr` strategy.

TLDR: validated on load and partially validated on dump (ignoring dynamic `setattr` attributes)